src/bun.js/test/jest.zig

```diff
@@ -2080,6 +2445,225 @@ pub const Expect = struct {
+            return .zero;
+        };
+
+        if (result) |saved_value| {
+            var pretty_value: MutableString = MutableString.init(default_allocator, 0) catch unreachable;
+            value.jestSnapshotPrettyFormat(&pretty_value, globalObject) catch {
+                globalObject.throw("Failed to pretty format value", .{});
+                return .zero;
+            };
+            defer pretty_value.deinit();
+
+            if (strings.eql(pretty_value.toOwnedSliceLeaky(), saved_value)) {
+                return thisValue;
+            }
+
+            const signature = comptime getSignature("toMatchSnapshot", "<green>expected<r>", false);
+            const fmt = signature ++ "\n\n{any}\n";
+            const diff_format = DiffFormatter{
+                .received_string = pretty_value.toOwnedSliceLeaky(),
+                .expected_string = saved_value,
+                .globalObject = globalObject,
+            };
+
+            if (Output.enable_ansi_colors) {
+                globalObject.throw(Output.prettyFmt(fmt, true), .{diff_format});
+                return .zero;
+            }
+            globalObject.throw(Output.prettyFmt(fmt, false), .{diff_format});
+            return .zero;
+        }
+
+        return thisValue;
+    }
+
+    pub const PropertyMatcherIterator = struct {
+        received_object: JSValue,
+        failed: bool,
+        i: usize = 0,
+
+        pub fn forEach(
+            globalObject: *JSGlobalObject,
+            ctx_ptr: ?*anyopaque,
+            key_: [*c]ZigString,
+            value: JSValue,
+            _: bool,
+        ) callconv(.C) void {
+            const key: ZigString = key_.?[0];
+            if (key.eqlComptime("constructor")) return;
+            if (key.eqlComptime("call")) return;
+
+            var ctx: *@This() = bun.cast(*@This(), ctx_ptr orelse return);
+            defer ctx.i += 1;
+            var received_object: JSValue = ctx.received_object;
+
+            if (received_object.get(globalObject, key.slice())) |received_value| {
+                if (JSC.Jest.ExpectAny.fromJS(value)) |_| {
+                    var constructor_value = JSC.Jest.ExpectAny.constructorValueGetCached(value) orelse {
+                        globalObject.throw("Internal consistency error: the expect.any(constructor value) was garbage collected but it should not have been!", .{});
+                        ctx.failed = true;
+                        return;
+                    };
+
+                    if (received_value.isCell() and received_value.isInstanceOf(globalObject, constructor_value)) {
+                        received_object.put(globalObject, &key, value);
+                        return;
+                    }
+
+                    // check primitives
+                    var constructor_name = ZigString.Empty;
+                    constructor_value.getNameProperty(globalObject, &constructor_name);
```

Comment:
can you add a TODO here to make it check if the constructor is actually Number || Boolean || String || BigInt in the bindings by reading it from the JSGlobalObject property which (I think) is publicly exported? This approach of reading the class name is brittle (`var f = () => {}; f[Symbol.toStringTag] = "Number"`) and slow

Reply:
yeah, I was unable to find this one. Added the todo